projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-row.component.ts

```diff
@@ -234,12 +237,9 @@ export class IgxGridFilteringRowComponent implements AfterViewInit {
         this.isKeyPressed = true;
 
         if (this.column.dataType === DataType.Boolean) {
-            if ((event.key === KEYS.ENTER || event.key === KEYS.SPACE || event.key === KEYS.SPACE_IE) &&
-            this.dropDownConditions.collapsed) {
-                this.toggleConditionsDropDown(this.inputGroupPrefix.nativeElement);
-                event.stopPropagation();
-                return;
-            } else if ((event.key === KEYS.ESCAPE || event.key === KEYS.ESCAPE_IE) && !this.dropDownConditions.collapsed) {
```

Comment:
Why we're not handling Esc key anymore? That way pressing the Esc key will directly close the filtering row instead of closing the dropdown.

Reply:
With my changes when dataType is Boolean and click on input or press Enter/Space the focus is moved to the dropdown. If we change the focus of drop down(for example click on input or click tab key) it will be closed so we do not need this check.